core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -39,6 +40,16 @@ import org.apache.spark.scheduler.cluster.ExecutorInfo
 import org.apache.spark.storage._
 import org.apache.spark.util.Utils.weakIntern
 
+/**
+ * Helper class for passing configuration options to JsonProtocol.
+ * We use this instead of passing SparkConf directly because it lets us avoid
+ * repeated re-parsing of configuration values on each read.
+ */
+private[spark] class JsonProtocolOptions(conf: SparkConf) {
```

Comment:
This is the same flagging pattern used in https://github.com/apache/spark/pull/39763, so these PRs will conflict. I've marked this PR as WIP pending discussion on how / whether we want to flag this change (perhaps I'm being overly conservative and we'd be okay with saying that third-party code directly consuming event logs should update itself to be robust to the new / changed format and not go out of our way to provide escape hatches for this use-case).